src/test/bin/quic_gtest.h

```diff
@@ -697,11 +694,19 @@ struct ValidateConnectionEventArgs {
     uint32_t Test;
     static ::std::vector<ValidateConnectionEventArgs> Generate() {
         ::std::vector<ValidateConnectionEventArgs> list;
-#if !defined(QUIC_DISABLE_0RTT_TESTS) && !defined(QUIC_USE_RAW_DATAPATH) // TODO: Fix openssl/XDP bug and enable this back
-        for (uint32_t Test = 0; Test < 3; ++Test)
-#else
-        for (uint32_t Test = 0; Test < 2; ++Test)
+        uint32_t TestCount = 3;
+
+#if !defined(QUIC_DISABLE_0RTT_TESTS) // TODO: Fix openssl/XDP bug and enable this back
+        uint32_t Length = sizeof(uint32_t);
+        uint32_t Features = 0;
+        MsQuic->GetParam(
+            nullptr,
+            QUIC_PARAM_GLOBAL_DATAPATH_FEATURES,
+            &Length,
+            &Features);
+        TestCount = !!(Features & CXPLAT_DATAPATH_FEATURE_RAW_SOCKET) ? 2 : 3;
```

Comment:
You have to do this in several places. How about we make a helper function to query either if a feature is supported, or specifically just if RawSocket is supported?